pkg/operator/event.go

```diff
@@ -29,17 +29,56 @@ import (
 	kwatch "k8s.io/client-go/pkg/watch"
 )
 
-type Event struct {
+type clusterEvent struct {
 	Type   kwatch.EventType
 	Object *cluster.Cluster
 }
 
+type poolEvent struct {
+	Type   kwatch.EventType
+	Object *cluster.Pool
+}
+
 type rawEvent struct {
 	Type   kwatch.EventType
 	Object json.RawMessage
 }
 
-func pollEvent(decoder *json.Decoder) (*Event, *unversioned.Status, error) {
+func pollClusterEvent(decoder *json.Decoder) (*clusterEvent, *unversioned.Status, error) {
+	re, status, err := pollEvent(decoder)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to poll cluster event. %+v", err)
+	}
+
+	ev := &clusterEvent{
+		Type:   re.Type,
+		Object: &cluster.Cluster{},
+	}
+	err = json.Unmarshal(re.Object, ev.Object)
+	if err != nil {
+		return nil, nil, fmt.Errorf("fail to unmarshal Cluster object from data (%s): %v", re.Object, err)
+	}
+	return ev, status, nil
+}
+
+func pollPoolEvent(decoder *json.Decoder) (*poolEvent, *unversioned.Status, error) {
```

Comment:
error messages in this function should reference "pool event" instead of "cluster event"